### PR TITLE
Add check serial length test

### PR DESCRIPTION
**Pull request by qingwangrh**
ID:2008103
Signed-off-by: qingwangrh <qinwang@redhat.com>

### CODE REVIEW

**Comment by qingwangrh** [comment]
python ConfigTest.py --testcase=block_check_serial --guestname=Win2019,RHEL.8.4.0 --driveformat=virtio_blk,virtio_scsi --iothread_scheme=roundrobin --nr_iothreads=2 --machine=q35,i440fx --clone=no --customsparams="vm_mem_limit = 12G\ncd_format=ide\nimage_aio=threads" 

(1/8) Host_RHEL.m8.u6.product_rhel.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.block_check_serial: FAIL: Can not find serial or uid on disks:['stg1', 'stg2'] (35.03 s)
 (2/8) Host_RHEL.m8.u6.product_rhel.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.block_check_serial.q35: FAIL: Can not find serial or uid on disks:['stg1', 'stg2'] (31.65 s)
 (3/8) Host_RHEL.m8.u6.product_rhel.qcow2.virtio_blk.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.block_check_serial: FAIL: Can not find serial or uid on disks:['stg1', 'stg2'] (97.26 s)
 (4/8) Host_RHEL.m8.u6.product_rhel.qcow2.virtio_blk.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.block_check_serial.q35: FAIL: Can not find serial or uid on disks:['stg1', 'stg2'] (103.92 s)
 (5/8) Host_RHEL.m8.u6.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.block_check_serial: FAIL: Can not find serial or uid on disks:['stg1', 'stg2'] (31.94 s)
 (6/8) Host_RHEL.m8.u6.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.block_check_serial.q35: FAIL: Can not find serial or uid on disks:['stg1', 'stg2'] (29.92 s)
 (7/8) Host_RHEL.m8.u6.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.block_check_serial: FAIL: Can not find serial or uid on disks:['stg1', 'stg2'] (97.24 s)
 (8/8) Host_RHEL.m8.u6.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.block_check_serial.q35: FAIL: Can not find serial or uid on disks:['stg1', 'stg2'] (78.48 s)

The failed case is expected due to Bug 1967668 - Don't silently truncate (SCSI) disk serial number

File: qemu/tests/cfg/block_check_serial.cfg [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    image_serial_stg2="stg2_serial_123456789a_123456789b_123456789b_123456789d_123456789e_end"
+    image_serial_stg3="stg3_device_123456789a_123456789b_123456789b_123456789d_123456789e_end"
+    blk_extra_params_stg1 = "serial=${image_serial_stg1}"
+    blk_extra_params_stg2 = "serial=${image_serial_stg2}"
+    blk_extra_params_stg3 = "device_id=${image_serial_stg3}"
+    Windows:
+        cmd = "powershell -command "get-disk|? {{($_.UniqueId -eq '{0}') -or "
+        cmd += "($_.SerialNumber -eq '{0}')}}|FT -HideTableHeaders"
+    Linux:
+        cmd = "sg_vpd {0} -p 0x80 |tail -n 1 > {1};sg_vpd {0} -p 0x83 |tail -n 1 >> {1};"
```
Comment (zhenyzha) [inline qemu/tests/cfg/block_check_serial.cfg:29]: Could we use a more accurate cmd name?
like: `cmd_get_disk_id`

File: qemu/tests/block_check_serial.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+    def _find_disks_by_serial():
+        """
+        Find the disk name by image serial in guest.
+        """
+        wrong_disks = []
+        for img in images:
+            image_params = params.object_params(img)
+            serial = image_params['image_serial']
+            logging.info("Try to Find img %s by %s", img, serial)
```
Comment (zhenyzha) [inline qemu/tests/block_check_serial.py:32]: Try to find the image

Comment (qingwangrh) [reply qemu/tests/block_check_serial.py]: updated

File: qemu/tests/cfg/block_check_serial.cfg (hunk 1 above)
Comment (qingwangrh) [reply qemu/tests/cfg/block_check_serial.cfg]: updated

**Comment by zhencliu** [comment]
Suggest to update commit msg:
block_check_serial: Add check serial length test
Starting with the case name, which can be explicit

Others LGTM

**Comment by qingwangrh** [comment]
> Suggest to update commit msg:
> block_check_serial: Add check serial length test
> Starting with the case name, which can be explicit
> 
> Others LGTM
updated

**Pull request merged**